// stoker/dist/cjs/openapi/schemas/create-error-schema.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _zodOpenapi = require("@hono/zod-openapi");
const createErrorSchema = schema => {
  const {
    error
  } = schema.safeParse(schema._def.type === "array" ? [schema.element._def.type === "string" ? 123 : "invalid"] : {});
  const example = error ? {
    name: error.name,
    issues: error.issues.map(issue => ({
      code: issue.code,
      path: issue.path,
      message: issue.message
    }))
  } : {
    name: "ZodError",
    issues: [{
      code: "invalid_type",
      path: ["fieldName"],
      message: "Expected string, received undefined"
    }]
  };
  return _zodOpenapi.z.object({
    success: _zodOpenapi.z.boolean().openapi({
      example: false
    }),
    error: _zodOpenapi.z.object({
      issues: _zodOpenapi.z.array(_zodOpenapi.z.object({
        code: _zodOpenapi.z.string(),
        path: _zodOpenapi.z.array(_zodOpenapi.z.union([_zodOpenapi.z.string(), _zodOpenapi.z.number()])),
        message: _zodOpenapi.z.string().optional()
      })),
      name: _zodOpenapi.z.string()
    }).openapi({
      example
    })
  });
};
module.exports = createErrorSchema;
